### PR TITLE
Add note about the planned change of the manylinux default

### PR DESCRIPTION
**Pull request by joerick**
As per discussion in #1772, note about the planned update of the manylinux default.

### CODE REVIEW

File: docs/options.md [hunk 1]
```diff
@@ -1157,25 +1157,30 @@ Platform-specific environment variables are also available:<br/>
...
+| CIBW_MANYLINUX_S390X_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_s390x) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_s390x)
+| CIBW_MANYLINUX_PYPY_AARCH64_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_aarch64) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_aarch64)
+| CIBW_MANYLINUX_PYPY_I686_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_i686) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_i686)
+| CIBW_MUSLLINUX_X86_64_IMAGE | [`musllinux_1_2`](https://quay.io/pypa/musllinux_1_2_x86_64) |
+| CIBW_MUSLLINUX_I686_IMAGE | [`musllinux_1_2`](https://quay.io/pypa/musllinux_1_2_i686) |
+| CIBW_MUSLLINUX_AARCH64_IMAGE | [`musllinux_1_2`](https://quay.io/pypa/musllinux_1_2_aarch64) |
+| CIBW_MUSLLINUX_PPC64LE_IMAGE | [`musllinux_1_2`](https://quay.io/pypa/musllinux_1_2_ppc64le) |
+| CIBW_MUSLLINUX_S390X_IMAGE | [`musllinux_1_2`](https://quay.io/pypa/musllinux_1_2_s390x) |
+
+<small>* The default is scheduled to change in a cibuildwheel release on or after 30th June 2025 - it's recommended to set the value now to avoid this changing by itself.</small>
```
Comment (joerick) [inline docs/options.md:1181]: Pending decision in #1772

Comment (henryiii) [reply docs/options.md:1181]:
```suggestion
<small>* The default is scheduled to change in a cibuildwheel release on or after 30th June 2025 - if you don't want the new default, you should set the value to `manylinux2014`.</small>
```

I think many projects are happy to accept whatever is default, so this wording is a bit softer.

Comment (joerick) [reply docs/options.md]: Thanks, yeah that's better

**Comment by Czaki** [comment]
Maybe it will be a good practice to promise that all changing manylinux images should be end of January?

**Comment by joerick** [comment]
> Maybe it will be a good practice to promise that all changing manylinux images should be end of January?

I'm afraid I don't follow your logic. What's the relevance of January? The proposal was to change the default at the end of June 2025.

**Comment by Czaki** [comment]
> I'm afraid I don't follow your logic. What's the relevance of January? The proposal was to change the default at the end of June 2025.

I do not know why I read June as January. 
In general, when I read in `black` documentation, that they promise not to introduce changes in style guide except begin of the year. And maybe having information that a change of some default value will happen only at given month may simplify life.

**Comment by joerick** [comment]
Oh, I see. Yeah, that's a good idea. We're often beholden to external factors too, but let's see if it works to stick to a June cadence on the next one.

**Comment by joerick** [comment]
@henryiii @mayeut , approval/review here would be good. Test fail is Travis and unrelated.

**Review by henryiii — APPROVED** [review]
Fine by me, but mostly needs @mayeut's approval.

**Review by mayeut — CHANGES_REQUESTED** [review]
> Oh, I see. Yeah, that's a good idea. We're often beholden to external factors too, but let's see if it works to stick to a June cadence on the next one.

Indeed external factors do exist and not only for cibuildwheel.
In order for the image sizes of manylinux not to grow too much, I'll try to remove EOL ones (when they can be removed) at the time of the new Python 3.x beta.
Given the release schedule of last CPython versions, this generally happens at the beginning of May.

File: docs/options.md [hunk 2]
```diff
@@ -1157,26 +1157,31 @@ Platform-specific environment variables are also available:<br/>
...
-Set an alternative Docker image to be used for building [manylinux / musllinux](https://github.com/pypa/manylinux) wheels.
+
+> Specify manylinux / musllinux container images
+
+The available options are:
+
+| Option | Default | Future default* |
+|---|---|---|
+| CIBW_MANYLINUX_X86_64_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_x86_64) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_x86_64)
+| CIBW_MANYLINUX_I686_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_i686) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_i686)
```
Comment (mayeut) [inline docs/options.md:1168]:
```suggestion
| CIBW_MANYLINUX_I686_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_i686) | 
```

There's no i686 image for manylinux_2_28

File: docs/options.md [hunk 3]
```diff
@@ -1157,26 +1157,31 @@ Platform-specific environment variables are also available:<br/>
...
+| CIBW_MANYLINUX_PYPY_X86_64_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_x86_64) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_x86_64)
+| CIBW_MANYLINUX_AARCH64_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_aarch64) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_aarch64)
+| CIBW_MANYLINUX_PPC64LE_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_ppc64le) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_ppc64le)
+| CIBW_MANYLINUX_S390X_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_s390x) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_s390x)
+| CIBW_MANYLINUX_PYPY_AARCH64_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_aarch64) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_aarch64)
+| CIBW_MANYLINUX_PYPY_I686_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_i686) | [`manylinux_2_28`](https://quay.io/pypa/manylinux_2_28_i686)
```
Comment (mayeut) [inline docs/options.md:1174]:
```suggestion
| CIBW_MANYLINUX_PYPY_I686_IMAGE | [`manylinux2014`](https://quay.io/pypa/manylinux2014_i686) | 
```
There's no i686 image for manylinux_2_28

**Review by mayeut — APPROVED** [review]

**Pull request merged**